### PR TITLE
Italian translation

### PR DESCRIPTION
**Pull request by Hissvard**
I'd like to point out I was getting a '"it" is not a valid locale' message when running it on my machine and had to edit and add `it` to these files in order to eliminare the 500 error I was getting:
 * config/application.rb
 * spec/support/compare_locales_support.rb

However, I could not find mention of these files having to be modified in the Translations guide. I have zero experience with Ruby so I'd like to know whether this is the way it's meant to be done or I did something wrong.

I also know it's better to have more than a pair of eyes when translating to be sure the translation feels natural and is correct, but I honestly don't know anyone I could ask this to - it'd be nice if someone else could review it as I'm sure some of the wording is off.

### CODE REVIEW

**Comment by julianguyen** [comment]
Hey @Hissvard, thank you so much for translating our app to Italian! :) It's much appreciated! Can you email us at join.ifme@gmail.com so we can add you to our Slack?

File: app/assets/javascripts/announcement.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ function scrollToLocaleToggle() {
 
 function translationsAnnouncement() {
   if ($("body").hasClass("pages home") && $("html").attr("lang") === "en") {
-    $("#announcement").html("More communities need mental health support. Our site is now available in <span id=\"translations_anchor\" class=\"anchor\">Español, Nederlands, Português and Svenska</span>!");
+    $("#announcement").html("More communities need mental health support. Our site is now available in <span id=\"translations_anchor\" class=\"anchor\">Español, Nederlands, Português, Svenska and Italian</span>!");
```
Comment (sexybiggetje) [inline app/assets/javascripts/announcement.js]: Please alphabetize as: Español, Italian, Nederlands, Português and Svenska.
Also these names are localized, shouldn't it become Italiano?

**Comment by sexybiggetje** [comment]
There are some keys missing in your it.yml;

> Processing: en.yml vs it.yml
> ----
> Key doesn't exist: it.draft
> Key doesn't exist: it.languages.nl
> Key doesn't exist: it.common.actions.submit_publish
> Key doesn't exist: it.common.actions.submit_draft
> Key doesn't exist: it.moments.form.draft_question
> Key doesn't exist: it.strategies.form.draft_question

I also wrote this tool to provide spelling suggestions and style advice, ignore the errors about %{} variables, but it still provides a big list. Can you review the suggestions in the attachement?
[suggestlang-it.txt](https://github.com/julianguyen/ifme/files/1436657/suggestlang-it.txt)

File: app/assets/javascripts/announcement.js (hunk 1 above)
Comment (Hissvard) [reply app/assets/javascripts/announcement.js]: Thank you, really hadn't noticed that

File: config/locales/devise.it.yml [hunk 2]
```diff
@@ -15,8 +15,8 @@ it:
       last_attempt: "Ti rimane solo un altro tentativo prima che il tuo account venga bloccato."
       not_found_in_database: "%{authentication_keys} o password invalidi."
       timeout: "La tua sessione è scaduta. Per favore, effettua nuovamente il login per continuare."
-      unauthenticated: "&Egrave; necessario effettuare l’accesso o la registrazione prima di continuare."
-      unconfirmed: "&Egrave; necessario confermare il tuo indirizzo E-Mail prima di continuare."
+      unauthenticated: "È necessario effettuare l’accesso o la registrazione prima di continuare."
+      unconfirmed: "È necessario confermare il tuo indirizzo EMail prima di continuare."
```
Comment (sexybiggetje) [inline config/locales/devise.it.yml]: How about Email? Instead of EMail?

File: config/locales/devise.it.yml [hunk 3]
```diff
@@ -30,26 +30,26 @@ it:
       failure: "Non è stato possibile autenticarti da %{kind} perché \"%{reason}\"."
       success: "Autenticazione effettuata con successo da %{kind}."
     passwords:
-      no_token: "Puoi solo accedere a questa pagina tramite una mail di reset password. Se l’hai effettivamente aperta da una mail di reset, assicurati di aver copiato l’url completo."
-      send_instructions: "Riceverai una mail con istruzioni su come resettare la password in alcuni minuti."
-      send_paranoid_instructions: "Se il tuo indirizzo E-Mail esiste, riceverai una mail contenente le istruzioni necessarie a resettare la tua password in alcuni minuti."
+      no_token: "Puoi solo accedere a questa pagina tramite una Email di reset password. Se l’hai effettivamente aperta da una Email di reset, assicurati di aver copiato l’url completo."
+      send_instructions: "Riceverai una Email con istruzioni su come resettare la password in alcuni minuti."
+      send_paranoid_instructions: "Se il tuo indirizzo EMail esiste, riceverai una Email contenente le istruzioni necessarie a resettare la tua password in alcuni minuti."
```
Comment (sexybiggetje) [inline config/locales/devise.it.yml]: How about Email? Instead of EMail?

File: config/locales/devise.it.yml [hunk 4]
```diff
@@ -30,26 +30,26 @@ it:
...
       updated: "La tua password è stata modificata con successo. Sei ora loggato."
       updated_not_active: "La tua password è stata modificata con successo."
     registrations:
       destroyed: "Arrivederci! Il tuo account è stato eliminato con successo. Speriamo di rivederti presto."
       signed_up: "Benvenuto! Registrazione effettuata con successo."
-      signed_up_but_inactive: "Ti sei registrato con successo. Non siamo però riusciti a loggarti perché il tuo indirizzo E-Mail non è ancora stato confermato."
+      signed_up_but_inactive: "Ti sei registrato con successo. Non siamo però riusciti a loggarti perché il tuo indirizzo EMail non è ancora stato confermato."
```
Comment (sexybiggetje) [inline config/locales/devise.it.yml]: And again same here :)

File: config/locales/devise.it.yml [hunk 5]
```diff
@@ -30,26 +30,26 @@ it:
...
       signed_up_but_locked: "Ti sei registrato con successo. Non siamo però riusciti a loggarti perché il tuo account è bloccato."
-      signed_up_but_unconfirmed: "Ti è stata inviata una mail contenente il link per completare la registrazione. Per favore, clicca sul link per completare la tua registrazione."
-      update_needs_confirmation: "Hai aggiornato il tuo account con successo, ma dobbiamo verificare il tuo indirizzo E-Mail. Per favore, controlla la tua casella e clicca sul link per confermare il nuovo indirizzo."
+      signed_up_but_unconfirmed: "Ti è stata inviata una Email contenente il link per completare la registrazione. Per favore, clicca sul link per completare la tua registrazione."
+      update_needs_confirmation: "Hai aggiornato il tuo account con successo, ma dobbiamo verificare il tuo indirizzo EMail. Per favore, controlla la tua casella e clicca sul link per confermare il nuovo indirizzo."
```
Comment (sexybiggetje) [inline config/locales/devise.it.yml]: And this EMail as well?

**Comment by Hissvard** [comment]
I noticed other locales also used the Email/email form so I joined the club, but now that you made me notice EMail does indeed have a much nicer shape!

**Comment by Hissvard** [comment]
As a side note I did try to add tests for Italian to `compare_locales_support_spec.rb` but I sadly don't know Ruby so it seems my attempt failed.

**Comment by sexybiggetje** [comment]
I think the check fails because there are still some keys missing i it.yml, see my list above :)

**Comment by sexybiggetje** [comment]
Hmm, missing keys have been resolved. @baohouse has helped me a lot, maybe he has an idea about this check?

**Comment by baohouse** [comment]
Yah, sorry @Hissvard, the error messages aren't very clear. It looks like something is missing from nl.yml (dutch translation file). My guess is that "it: Italiano" is missing from that file under the "languages" tree.

**Comment by Hissvard** [comment]
Daaaaamn, never would have thought about that. Thanks a ton to both!

**Comment by Hissvard** [comment]
I'd have someone else review but I don't have in mind anyone who could lend me a pair of eyes for checking this, sadly. Think we could send out a tweet?

**Comment by timofonic** [comment]
@Hissvard Grazie! (I know a bit of Italian and understand it easily because I'm Spanish)

@julianguyen  @baohouse @sexybiggetje Any news about this? I want to translate it to Spanish, but I want to see other translations merged first.

**Comment by Hissvard** [comment]
Well, it's not like there's something terribly wrong hidden somewhere in the translation, just thought having more than a couple eyes review it would be ideal - we can merge this if it's okay with everyone.

**Comment by sexybiggetje** [comment]
I've tweeted about this a few hours ago, some OSS accounts picked it up.
https://twitter.com/bexysiggetje/status/929683005093924869

You can review the files if you want @timofonic, take a look at the [translation page](https://github.com/julianguyen/ifme/wiki/Translations) on the wiki. I've written a small bit for finishing up.

The project usually requires another set of eyes for quality assurance. Correct language is important. If I reflect on how much small typos I've made during the Dutch translation, I'm glad that there were a few reviewers :). I'm in favour of merging, but really let's wait a small bit for reviewers. And take a look at the suggestions file I've attached above generated by the tool. It's automated reviewing ;).

**Comment by julianguyen** [comment]
@Hissvard Feel free to merge whenever you're ready! Anyone who wants to update the translations can do it after anyways :P

**Pull request merged**